include/cxxopts.hpp

```diff
@@ -761,19 +764,40 @@ namespace cxxopts
         return !result.empty();
       }
 
-      inline std::pair<std::string, std::string> SplitSwitchDef(const std::string &text)
+      // Gets the option names specified via a single, comma-separated string,
+      // and returns the separate, space-discarded, non-empty names
+      // (without considering which or how many are single-character)
+      inline OptionNames split_option_names(const std::string &text)
       {
-        std::match_results<const char*> result;
-        std::regex_match(text.c_str(), result, option_specifier);
-        if (result.empty())
+        if (not std::regex_match(text.c_str(), option_specifier))
         {
           throw_or_mimic<invalid_option_format_error>(text);
         }
 
-        const std::string& short_sw = result[2];
-        const std::string& long_sw = result[3];
+        OptionNames split_names;
+        std::string::size_type token_start_pos = 0;
+        auto length = text.length();
 
-        return std::pair<std::string, std::string>(short_sw, long_sw);
+        // TODO: Try avoiding having to tokenize ourselves
```

Comment:
`std::regex_iterator`?

Reply:
std::token_regex_iterator will do, apparently. Thanks for the idea.